superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/dndControls.tsx

```diff
@@ -25,43 +26,73 @@ import {
   t,
   validateNonEmpty,
 } from '@superset-ui/core';
-import { ExtraControlProps, SharedControlConfig, Dataset } from '../types';
+import {
+  ExtraControlProps,
+  SharedControlConfig,
+  Dataset,
+  Metric,
+} from '../types';
 import { DATASET_TIME_COLUMN_OPTION, TIME_FILTER_LABELS } from '../constants';
-import { QUERY_TIME_COLUMN_OPTION, defineSavedMetrics } from '..';
+import {
+  QUERY_TIME_COLUMN_OPTION,
+  defineSavedMetrics,
+  ColumnOption,
+  ColumnMeta,
+  FilterOption,
+} from '..';
 import { xAxisControlConfig } from './constants';
 
-export const dndGroupByControl: SharedControlConfig<'DndColumnSelect'> = {
+type Control = {
+  savedMetrics?: Metric[] | null;
+  default?: unknown;
+};
+
+export const dndGroupByControl: SharedControlConfig<
```

Comment:
Could you add a comment that will explain the idea behind using props from non-dnd component in dnd control definition? Might be useful for other contributors in the future